docs/docs/for-developers/contributing.md

```diff
@@ -24,6 +24,65 @@ To improve the chances to get a pull request merged, you should select an issue
 
 In addition, Onivim 2 is built on [Revery](https://github.com/revery-ui/revery) - any work or improvements there will directly improve Onivim 2, as well!
 
+#### Changelog generation
+
+We use a script to generate the change log based on metadata picked up from the commit message, PR title and PR body. For the changelog to be generated successfully it's therefore important that these follow the exact format explained below.
+
+##### PR title
+
+PR titles should have the following format:
+
+```
+<type>(<scope>): <subject>
+```
+
+where `type` can be one of
+
+* **feat:** A new feature
+* **fix:** A bug fix
+* **refactor:** A code change that neither fixes a bug nor adds a feature
+* **perf:** A code change that improves performance
+* **test:** Adding missing or correcting existing tests
+* **docs:** Documentation only changes
+* **chore:** Changes to the build process or auxiliary tools and libraries such as documentation generation
+* **style:** Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
+
+`scope` is optional, but "encouraged". It is typically one of the "area" [issue labels](https://github.com/onivim/oni2/labels) or the name of a feature project converted to "humanized" title case. E.g. instead of `language-support` use `Language Support`, i.e. two words with the first letter of each word capitalized.
```

Comment:
```suggestion
`scope` is optional, but "encouraged". It is typically one of the "area" [issue labels](https://github.com/onivim/oni2/labels) or the name of a feature project.
```

I noticed you got this wrong on your first attempt @bryphe, so maybe we should just automatically convert `language-support`, `language support` and `LanguageSupport` to `Language Support` instead :wink:

Reply:
oops 😄 Yes, perhaps automatically changing it to 'capitalized human case' would be easiest